### PR TITLE
K8s Support

### PR DESCRIPTION
**Pull request by akutz**
This patch adds support for CSI's centralized model (ex. K8s) by performing Controller Attach/Detach logic inside of the Node service's Pub/Unpub RPCs.

### CODE REVIEW

File: service/node.go [hunk 1]
```diff
@@ -58,3 +311,30 @@ func (s *service) NodeGetCapabilities(
 		},
 	}, nil
 }
+
+func validateDevice(devicePath string) (string, error) {
```
Comment (codenrhoden) [inline service/node.go]: `gofsutil.ValidateDevice` is not usable here?

File: service/node.go [hunk 2]
```diff
@@ -1,25 +1,278 @@
...
+		}
+		lf["accessType"] = "mount"
+		lf["fsType"] = fsType
+		lf["mntFlags"] = mntFlags
+	default:
+		return nil, gocsi.ErrAccessTypeRequired
+	}
+
+	// Attach the VMDK to this node.
+	dev, err := s.ops.Attach(req.VolumeId, nil)
```
Comment (codenrhoden) [inline service/node.go:77]: I'm not familiar with how `vmdkops` works here, but will it gracefully handle if the device is already attached? I'm wondering if there is any code that can be short-circuited if the device is already attached. Thought maybe that isn't necessary, as that is left to the idempotent provider.

**Review by codenrhoden — APPROVED** [review]
Looks like this will take care of doing all the controller actions for attach/detach in the node pub/unpub requests instead. Looks good!

File: service/node.go (hunk 1 above)
Comment (akutz) [reply service/node.go]: Hi @codenrhoden,

It didn't exist when I wrote the above code. I added `ValidateDevice` to `gofsutil` later. I will update to the latest version of gofsutil and use its `ValidateDevice`. Good catch.

File: service/node.go (hunk 2 above)
Comment (akutz) [reply service/node.go:77]: Hi @codenrhoden,

The backend ESXi component's `attach` call is idempotent.

**Pull request merged**